### PR TITLE
Feature/table component

### LINKED ISSUE #85: Table component
As an Author, I want to have the possibility to add [Table](https://bulma.io/documentation/elements/table/) component.

**Assumtion**
The table is a little bit complicated component but let's start with something simple, and then we will be able to extend it.

Let's prepare the components:

- [ ] [Table container](https://bulma.io/documentation/elements/table/#table-container)
as a container with allowed Table.
- [ ] Table
as a container with allowed Table header, Table footer, Table body, Table row. Should allow applying styles defined in Bulma documentation. Initial content should have a sample table with a header, body, three rows, and three cells.
- [ ] Table header
as a container with allowed Table row.
- [ ] Table footer
as a container with allowed Table row.
- [ ] Table body
as a container with allowed Table row.
- [ ] Table row
as a container with allowed Table cell and Table header cell.
- [ ] Table cell
as a component with the possibility to define text, link, title, number of columns and rows.
- [ ] Table header cell
as a component with the possibility to define text, link, title, number of columns and rows.

### PR DESCRIPTION
**Pull request by gfazek**
## Description

This PR introduces the bulma table component. I had a lot of trouble with the placeholders for table row etc. and needed to use some css hacks in author.scss to make it somewhat work, so let me know if you have a more ideal solution or suggestions.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) labeled with `bug`
- [x] New feature (non-breaking change which adds functionality) labeled with `enhancement`
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the [code standards](../CONTRIBUTING.md) of this project.
- [ ] My change requires updating the documentation. I have updated the documentation accordingly.

### CODE REVIEW

File: application/backend/src/main/resources/apps/bulma/components/table/dialog/.content.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  "sling:resourceType": "wcm/dialogs/dialog",
+  "tabs": {
+    "sling:resourceType": "wcm/dialogs/components/tabs",
+    "generalTab": {
+      "sling:resourceType": "wcm/dialogs/components/tab",
+      "label": "General",
+      "isBordered": {
+        "sling:resourceType": "wcm/dialogs/components/checkbox",
+        "name": "isBordered",
+        "label": "Add borders to all the cells"
```
Comment (paweljozwicki) [inline application/backend/src/main/resources/apps/bulma/components/table/dialog/.content.json]: Let's consider changing those labels to be consistent with the last one. I mean something like `Cells borders` or something like that. And this more descriptive text can be added to the `description`.

File: application/backend/src/main/resources/apps/bulma/components/table/.content.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "isContainer": true,
+  "allowedComponents": [
+    "/apps/bulma/components/table/tablehead",
+    "/apps/bulma/components/table/tablefoot",
+    "/apps/bulma/components/table/tablebody"
```
Comment (paweljozwicki) [inline application/backend/src/main/resources/apps/bulma/components/table/.content.json]: It should have also the possibility to add tablerow.

File: application/backend/src/main/resources/apps/bulma/components/table/tablecell/dialog/.content.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        "name": "text",
+        "label": "Text",
+        "required": true
+      },
+      "colspan": {
+        "sling:resourceType": "wcm/dialogs/components/numberfield",
+        "name": "colspan",
+        "min": 1,
+        "max": 100,
+        "label": "Colspan"
```
Comment (paweljozwicki) [inline application/backend/src/main/resources/apps/bulma/components/table/tablecell/dialog/.content.json]: Authors can not understand colspan/rowspan as they're from HTML. Let's rename them to Columns and Rows.

File: application/backend/src/main/resources/apps/bulma/components/table/tableheadcell/dialog/.content.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        "name": "text",
+        "label": "Text",
+        "required": true
+      },
+      "colspan": {
+        "sling:resourceType": "wcm/dialogs/components/numberfield",
+        "name": "colspan",
+        "min": 1,
+        "max": 100,
+        "label": "Colspan"
```
Comment (paweljozwicki) [inline application/backend/src/main/resources/apps/bulma/components/table/tableheadcell/dialog/.content.json]: Same as in tablecell.

File: application/backend/src/main/resources/apps/bulma/components/table/dialog/.content.json (hunk 1 above)
Comment (gfazek) [reply application/backend/src/main/resources/apps/bulma/components/table/dialog/.content.json]: Fixed!

File: application/backend/src/main/resources/apps/bulma/components/table/.content.json (hunk 2 above)
Comment (gfazek) [reply application/backend/src/main/resources/apps/bulma/components/table/.content.json]: Fixed!

File: application/backend/src/main/resources/apps/bulma/components/table/tablecell/dialog/.content.json (hunk 3 above)
Comment (gfazek) [reply application/backend/src/main/resources/apps/bulma/components/table/tablecell/dialog/.content.json]: Fixed!

File: application/backend/src/main/resources/apps/bulma/components/table/tableheadcell/dialog/.content.json (hunk 4 above)
Comment (gfazek) [reply application/backend/src/main/resources/apps/bulma/components/table/tableheadcell/dialog/.content.json]: Fixed!

**Review by paweljozwicki — APPROVED** [review]

**Pull request merged**